src/MCPServer/lib/RPCServer.py

```diff
@@ -307,6 +312,118 @@ def _get_processing_config_fields_handler(self, worker, job):
         """
         return get_processing_fields(self.workflow)
 
+    def _units_statuses_handler(self, worker, job, payload):
+        """Returns the status of units that are of type SIP or Transfer.
+
+        It returns a JSON-encoded objects. Its ``objects`` attribute is an
+        array of objects, each of which represents a single unit. Each unit
+        has a ``jobs`` attribute shoe value is an array of objects, each of
+        which represents a job of the unit.
+
+        [config]
+        name = getUnitsStatuses
+        raise_exc = no
+        """
```

Comment:
This is the RPC that feeds the status in transfer/ingest. It used to be all done in the Dashboard. It was moved to RPCServer because it depends on the workflow data to build a comprehensive list of links and status for all the units being processed.

This is also good in the sense that the Dashboard becomes thinner and business logic is pushed to MCPServer.